Error when asking for 'points' stats without defining team value

Spotted error in the logs while trying to get `points` statistics:

```
[2021-08-20 11:26:53 +0000] [7] [ERROR] Error handling request /estatistikak?stat=points&league=EUSKOTREN
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/usr/lib/python3/dist-packages/gunicorn/workers/sync.py", line 176, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2301, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2287, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.6/dist-packages/flask_restx/api.py", line 638, in error_router
    return original_handler(f)
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1733, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/dist-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/local/lib/python3.6/dist-packages/flask_restx/api.py", line 636, in error_router
    return self.handle_error(e)
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 2284, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1807, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.6/dist-packages/flask_restx/api.py", line 638, in error_router
    return original_handler(f)
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1710, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.6/dist-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/usr/local/lib/python3.6/dist-packages/flask_restx/api.py", line 636, in error_router
    return self.handle_error(e)
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1805, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1791, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.6/dist-packages/flask_restx/api.py", line 375, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/flask/views.py", line 88, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/flask_restx/resource.py", line 44, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/opt/webapp/app/resources/estatistikak.py", line 303, in get
    result = EstatistikakLogic.get_points_per_race(league, year, category)
  File "/opt/webapp/app/resources/estatistikak.py", line 107, in get_points_per_race
    points_max = len(sailkapena['stats'])
TypeError: 'NoneType' object is not subscriptable
INFO:root:League:euskotren and year: None
```

--- a/app/resources/estatistikak.py
+++ b/app/resources/estatistikak.py
@@ -104,19 +104,20 @@ class EstatistikakLogic():
             league,
             year)
         estropadak = [estropada for estropada in estropadak if not estropada['izena'].startswith('Play')]
-        points_max = len(sailkapena['stats'])
-        for taldea, stats in sailkapena['stats'].items():
-            team_values = {
-                "key": taldea,
-                "color": get_team_color(taldea),
-            }
-            values = [{
-                "label": val[1]['izena'],
-                "x": i,
-                "value": points_max - val[0] + 1}
-                for i, val in enumerate(zip(stats['positions'], estropadak))]
-            team_values["values"] = values
-            result.append(team_values)
+        if sailkapena:
+            points_max = len(sailkapena['stats'])
+            for taldea, stats in sailkapena['stats'].items():
+                team_values = {
+                    "key": taldea,
+                    "color": get_team_color(taldea),
+                }
+                values = [{
+                    "label": val[1]['izena'],
+                    "x": i,
+                    "value": points_max - val[0] + 1}
+                    for i, val in enumerate(zip(stats['positions'], estropadak))]
+                team_values["values"] = values
+                result.append(team_values)
         return result
 
     @staticmethod
@@ -171,6 +172,8 @@ class EstatistikakLogic():
         result = []
         if team is None:
             sailkapena = EstatistikakDAO.get_sailkapena_by_league_year(league, year, None)
+            if not sailkapena or not sailkapena.get('stats', {}).get('age'):
+                return result
             min_ages = {
                 "key": 'Min',
                 "values": [{
@@ -300,10 +303,20 @@ class Estatistikak(Resource):
             if team:
                 result = EstatistikakLogic.get_points(league, team)
             else:
+                if not league or not year:
+                    return {"message": "You need to specify year and league"}, 400
                 result = EstatistikakLogic.get_points_per_race(league, year, category)
         elif stat_type == 'rank':
+            if team and not league:
+                return {"message": "You need to specify a league"}, 400
+            if not team and (not league or not year):
+                return {"message": "You need to specify year and league"}, 400
             result = EstatistikakLogic.get_rank(league, year, team, category)
         elif stat_type == 'ages':
+            if team and not league:
+                return {"message": "You need to specify a league"}, 400
+            if not team and (not league or not year):
+                return {"message": "You need to specify year and league"}, 400
             result = EstatistikakLogic.get_ages(league, year, team)
         elif stat_type == 'incorporations':
             result = EstatistikakLogic.get_incorporations(league, year, team)

